test/BUILD.gn

```diff
@@ -491,12 +491,31 @@ test("brave_unit_tests") {
   if (is_android) {
     deps += [
       ":brave_public_test_apk",
+      "//android_webview/browser",
```

Comment:
why are all of these new deps needed?

Reply:
cc: @samartnik

Reply:
Otherwise there are jni related errors like 
```
ld.lld: error: undefined symbol: Java_J_N_MfTomTpD
>>> referenced by brave_unit_tests__library__jni_registration_generated.cc:25252 (gen/jni_headers/brave/test/brave_unit_tests__library__jni_registration_generated.cc:25252)
>>>               obj/brave/test/brave_unit_tests__library/libbrave_unit_tests__library.so.lto.brave_unit_tests__library__jni_registration_generated.o:(Java_J_N__1V_1JJIJ)
```

Reply:
Actually some of them not required anymore, I did some clean up https://github.com/brave/brave-core/commit/716951531cb1bdd2c71d6ce92034d395fb984070